doc/modules/neural_networks_supervised.rst

```diff
@@ -0,0 +1,358 @@
+to the positive class, and the rest to the negative class.
+
+If there are more than two classes, :math:`f(x)` would instead pass through
+the softmax function, which is written as,
+
+.. math::
+      \text{Softmax} = \frac{\exp(W_i^Tx)}{\sum_{l=1}^k\exp(W_l^Tx)} 
+
+where :math:`W_i` contains the weights incident to the output neuron representing
+class :math:`i`, and :math:`K` is the number of classes. The result is a vector 
+containing the probabilities that sample :math:`x` belong to each class. The 
+output is the class with the highest probability.
+
+In regression, the output remains as :math:`f(x)`; therefore, output activation
+function is just the identity function.
+
+MLP uses different loss functions depending on the problem type. The loss 
+function for classification is Cross-Entropy, which in binary case is given as,
+
+.. math::
+
+    Loss(x,y,W) = -y \ln {f(x)}+(1-y)\ln{(1-f(x))} + \alpha ||W||_2^2
+
+where :math:`\alpha ||W||_2^2` is an L2-norm regularization term (aka penalty)
+that penalizes model complexity; and :math:`\alpha > 0` is a non-negative 
+hyperparameter that controls the magnitude of the penalty term.
+
+For regression, MLP uses the Square Error loss function; written as,
+
+.. math::
+
+    Loss(x,y,W) = \frac{1}{2}||f(x) - y ||_2^2 + \alpha ||W||_2^2
+
+
+Starting from initial random weights, multi layer perceptron (MLP) minimizes 
+the loss function by repeatedly updating these weights. After computing
+the loss function value, a backward pass propagates it from the output layer 
+to the initial layer, providing each weight parameter with an update value 
+meant to decrease the loss function.
+
+In gradient descent, the weight gradient :math:`\nabla W_{loss}` with respect
+to the loss function is computed and added to the corresponding :math:`W`.
+More formally, this is expressed as,
+
+.. math::
+
+    W^{i+1} = W^i + \epsilon \nabla W^i_{loss}
+
+
+where :math:`i` is the iteration step, and :math:`\epsilon` is the learning rate 
+with a value ranging from 0 and 1. 
+
+The algorithm stops when it reaches the set number of iterations; or 
+when the loss value is below a certain, small number.
+
+
+
+.. _mlp_tips:
+
+Tips on Practical Use
+=====================
+
+  * Multi-layer Perceptron is sensitive to feature scaling, so it
+    is highly recommended to scale your data. For example, scale each
+    attribute on the input vector X to [0, 1] or [-1, +1], or standardize
+    it to have mean 0 and variance 1. Note that you must apply the *same* 
+    scaling to the test vector for meaningful results. 
+    You can use :class:`StandardScaler`:: for standarization.
+
+      from sklearn.preprocessing import StandardScaler
+      scaler = StandardScaler()
+      scaler.fit(X_train)  # Don't cheat - fit only on training data
+      X_train = scaler.transform(X_train)
+      X_test = scaler.transform(X_test)  # apply same transformation to test data
```

Comment:
Maybe it should be mentioned that putting a `StandardScaler` in a `Pipeline` object which is fit to training data takes care of this transparently and gracefully